src/test/java/picard/illumina/IlluminaBasecallsToSamAdapterClippingTest.java

```diff
@@ -52,22 +56,26 @@ public String getCommandLineProgramName() {
      * Run IlluminaBasecallsToSam on a few test cases, and verify that results agree with hand-checked expectation.
      */
     @Test(dataProvider="data")
-    public void testBasic(final String LANE, final String readStructure) throws Exception {
+    public void testBasic(final String LANE, final String readStructure, final String fivePrimerAdapter, final String threePrimerAdapter) throws Exception {
         // Create the SAM file from Gerald output
         final File samFile = File.createTempFile("." + LANE + ".illuminaBasecallsToSam", ".sam");
         samFile.deleteOnExit();
-        final String[] illuminaArgv = {
-                "BASECALLS_DIR=" + TEST_DATA_DIR,
+        final List<String> illuminaArgv = CollectionUtil.makeList("BASECALLS_DIR=" + TEST_DATA_DIR,
                 "LANE=" + LANE,
                 "RUN_BARCODE=" + RUN_BARCODE,
                 "READ_STRUCTURE=" + readStructure,
                 "OUTPUT=" + samFile,
                 "ALIAS=" + ALIAS
-        };
+        );
+        if (fivePrimerAdapter != null) {
+            illuminaArgv.addAll(CollectionUtil.makeList(
+                "ADAPTERS_TO_CHECK=null",
+                "FIVE_PRIME_ADAPTER=" + fivePrimerAdapter,
+                "THREE_PRIME_ADAPTER=" + threePrimerAdapter
+            ));
+        }
         Assert.assertEquals(runPicardCommandLine(illuminaArgv), 0);
 
-        System.out.println ("Ouput Sam file is in " + samFile.getAbsolutePath());
-
         // Read the file and confirm it contains what is expected
         final SamReader samReader = SamReaderFactory.makeDefault().open(samFile);
```

Comment:
The test will not fail when fivePrimerAdapter!=null && threePrimerAdapter!=null even if there are clipped adaptors, also it will not fail if there are never any clipped adaptors....

Can you add "expected number of clipped reads" to the test input and verify that they right number got clipped?